client/src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import FocusLock from "react-focus-lock";
+
+import "./Sidebar.scss";
+
+import { CSSTransition } from "react-transition-group";
+
+import { trivial_text_maker } from "src/models/text";
+
+import { IconX } from "src/icons/icons";
+
+interface SidebarProps {
+  is_open: boolean;
+  callback: (value: boolean) => void;
+  children: React.ReactElement;
+  title_text: string;
+  sidebar_toggle_target: string;
+  return_focus_target: HTMLElement | undefined;
+  keydown_close: () => void;
+  keydown_close_value: boolean;
```

Comment:
Best I can tell, this pair of props just exist so that the state for `keydown_close_value` can be owned by the `GlossarySidebarController`. In the process, it creates a lot of tricky to following passing down (and then via the call back, up) of props/information across three different components. Ouch! Really complicates this particular use case, and will definitely make future use/maintenance of `Sidebar` a challenge.

So, in case I missed something, the `Sidebar` itself could just manage this all internally, right? If I'm right then I think I've got some follow up advice, so let me know and we can talk further :+1:

Reply:
Attempted a different approach to keep this all internal, let me know if that works!